classifier_params[] update_params
int32 selected_centroid
int32 filter_method
